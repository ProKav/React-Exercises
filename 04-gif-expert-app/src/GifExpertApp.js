import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
	const [categories, setCategories] = useState(defaultCategories);
	/* const [categories, setCategories] = useState(['Coding']) */

	return (
		<>
			<h2>Gift Expert App</h2>
			<h3>Input text for search</h3>
			<br />
			<AddCategory setCategories={setCategories} />
			<br />
			{categories.map((category) => (
				<GiftGrid key={category} category={category} />
			))}
		</>
	);
};

export default GifExpertApp;
