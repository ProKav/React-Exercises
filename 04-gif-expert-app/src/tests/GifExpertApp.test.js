import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas para el componente <GifExpertApp />', () => {
	test('Debe hacer un snapshoot y mostrarse correctamente', () => {
		const wrapper = shallow(<GifExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe mostrar una lista de categorias', () => {
		const categories = ['Coding', 'Dragon Ball GT'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GiftGrid').length).toBe(categories.length);
	});
});
