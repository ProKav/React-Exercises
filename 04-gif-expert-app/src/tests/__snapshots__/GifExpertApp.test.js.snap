// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifExpertApp /> Debe hacer un snapshoot y mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <h3>
    Input text for search
  </h3>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
</Fragment>
`;

exports[`Pruebas para el componente <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <h3>
    Input text for search
  </h3>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <GiftGrid
    category="Coding"
    key="Coding"
  />
  <GiftGrid
    category="Dragon Ball GT"
    key="Dragon Ball GT"
  />
</Fragment>
`;
